src/program.c

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#define _GNU_SOURCE
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+
+#if defined _WIN32 || defined __CYGWIN__
+#include <Windows.h>
+#else
+#include <unistd.h>
+#endif
+
+#include "rcutils/allocator.h"
+#include "rcutils/error_handling.h"
+
+int rcutils_get_pid(void)
+{
+#if defined _WIN32 || defined __CYGWIN__
+  return (int)GetCurrentProcessId();
+#else
+  return (int)getpid();
+#endif
+}
+
+char *rcutils_get_program_name(rcutils_allocator_t allocator)
+{
+  RCUTILS_CHECK_ALLOCATOR_WITH_MSG(
+    &allocator, "invalid allocator", return NULL);
```

Comment:
I don't think this needs to be wrapped.

Reply:
I don't feel really strongly about it either way, but doing the check here is consistent with what we do elsewhere.  Why do you think we don't need this particular check?

Reply:
I think the check is fine, I just meant it didn't need to be wrapped onto two lines. It's fine as-is.